app/src/main/java/blockeq/com/stellarwallet/services/networking/Horizon.kt

```diff
@@ -98,17 +98,56 @@ class Horizon {
 
             override fun onPostExecute(result: ErrorResponse?) {
                 if (result != null) {
-                    listener.onSendError()
+                    listener.onError()
                 } else {
-                    listener.OnSendSuccess()
+                    listener.onSuccess()
                 }
             }
         }
 
+        class JoinInflationDestination(private val listener: SuccessErrorCallback,
+                                       private val inflationDest : String)
+            : AsyncTask<KeyPair, Void, ErrorResponse>() {
+
+            override fun doInBackground(vararg params: KeyPair?): ErrorResponse? {
+                Network.usePublicNetwork()
+
+                val server = Server(PROD_SERVER)
+                val sourceKeyPair = params[0]
+                val sourceAccount = server.accounts().account(sourceKeyPair)
+                val destKeyPair = KeyPair.fromAccountId(inflationDest)
+
+                try {
+                    val transaction = Transaction.Builder(sourceAccount)
+                            .addOperation(SetOptionsOperation.Builder()
+                                    .setInflationDestination(destKeyPair)
+                                    .build())
+                            .build()
+
+                    transaction.sign(sourceKeyPair)
+                    server.submitTransaction(transaction)
+
+                } catch (error : ErrorResponse) {
+                    Log.d(TAG, error.body.toString())
+                    return error
+                }
+                return null
+            }
+
+            override fun onPostExecute(result: ErrorResponse?) {
+                if (result != null) {
+                    listener.onError()
+                } else {
+                    listener.onSuccess()
+                }
+            }
+
```

Comment:
Nitpick - however, i've noticed some extra spacing in the code. This could be part of the polish to go back and check however, moving forward we could keep in mind the spacing so it doesn't become a backlog item.